### PR TITLE
Add traces for HLS providers

### PR DESCRIPTION
**Pull request by pepeiborra**
These traces can be useful to diagnose performance issues

https://user-images.githubusercontent.com/26626/104844700-9ff16400-58c9-11eb-8d6d-53e9694cc292.mov

### CODE REVIEW

File: ghcide/src/Development/IDE/Core/Tracing.hs [hunk 1]
```diff
@@ -1,11 +1,13 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE DataKinds #-}
+#include "ghc-api-version.h"
```
Comment (ndmitchell) [inline ghcide/src/Development/IDE/Core/Tracing.hs:3]: Should we drop this now we don't have ghc-lib support?

**Review by ndmitchell — APPROVED** [review]
Super cool!

File: ghcide/src/Development/IDE/Core/Tracing.hs (hunk 1 above)
Comment (ndmitchell) [reply ghcide/src/Development/IDE/Core/Tracing.hs:3]: Or do we still have ghc-lib support? I'm confused...

Comment (pepeiborra) [reply ghcide/src/Development/IDE/Core/Tracing.hs:3]: we still do, but it's not being tested so it might have bit rotten

**Pull request merged**